source/slang/slang-parameter-binding.cpp

```diff
@@ -351,6 +353,7 @@ struct UsedRangeSet : RefObject
     // Information on what ranges of "registers" have already
     // been claimed, for each resource type
     UsedRanges usedResourceRanges[kLayoutResourceKindCount];
+    UsedRanges usedResourceRangesInputAttachmentIndex[kLayoutResourceKindCount];
```

Comment:
This will be also a lot of memory waste if it will be mostly unused.

Reply:
I agree. This can be reduced to `UsedRanges usedResourceRangesInputAttachmentIndex` in all likleyhood since we know this resourceKind is only used for images (which have a specific layout resource kind.

I will do this in the next commit